### PR TITLE
Update provide to take an array of keys, per spec

### PR DESCRIPTION
**Pull request by willscott**
cc @ischasny

### CODE REVIEW

File: client/contentrouting.go [hunk 1]
```diff
@@ -27,10 +28,28 @@ func (c *ContentRoutingClient) Provide(ctx context.Context, key cid.Cid, announc
...
+func (c *ContentRoutingClient) ProvideMany(ctx context.Context, keys []multihash.Multihash) error {
+	keysAsCids := make([]cid.Cid, 0, len(keys))
+	for _, m := range keys {
+		keysAsCids = append(keysAsCids, cid.NewCidV0(m))
+	}
+	_, err := c.client.Provide(ctx, keysAsCids, 24*time.Hour)
+	return err
+}
+
+// Ready is part of the existing `ProvideMany` interface, but can be used more generally to determine if the routing client
```
Comment (guseggert) [inline client/contentrouting.go:44]: > Ready is part of the existing `ProvideMany` interface

What does this mean? I can't find mention of this anywhere else.

File: client/contentrouting.go [hunk 2]
```diff
@@ -27,10 +28,28 @@ func (c *ContentRoutingClient) Provide(ctx context.Context, key cid.Cid, announc
...
+// has a working connection.
+func (c *ContentRoutingClient) Ready() bool {
```
Comment (guseggert) [inline client/contentrouting.go:46]: Sorry maybe I'm out of the loop here, but what is this for? What connection state are we envisioning using here? Checking if the underlying HTTP client has any conns? And how is this method going to be used by callers?

**Review by guseggert — CHANGES_REQUESTED** [review]

File: client/contentrouting.go (hunk 1 above)
Comment (willscott) [reply client/contentrouting.go:44]: https://github.com/ipfs/kubo/blob/9c6094e651bd30a11ee7a501bf9aa7eb5cc34deb/routing/wrapper.go#L35-L38

File: client/contentrouting.go (hunk 2 above)
Comment (willscott) [reply client/contentrouting.go:46]: i couldn't find documentation, but the current logic in kubo does an interface cast expecting this method to exist

Comment (guseggert) [reply client/contentrouting.go:46]: Ah okay thanks, I found it: https://github.com/ipfs/kubo/blob/master/routing/wrapper.go#L35

Looks like it's by the reprovider for the acclerated DHT client, to keep it from providing records until the routing table is bootstrapped.

Seems like a blocker for getting the accelerated DHT client working behind Reframe, but I think we can figure that out later.

**Review by guseggert — APPROVED** [review]

**Pull request merged**